addons/account/models/account_move.py

```diff
@@ -2421,9 +2421,6 @@ def copy_data(self, default=None):
                     for command, _id, line_vals in data['line_ids']
                     if command == Command.CREATE
                 ]
-            elif move.move_type == 'entry':
-                if 'partner_id' not in data:
```

Comment:
I would keep this `if` statement, in case we actually don't have data['partner_id'], it would be better to set it to `False` 
```diff
-                if 'partner_id' not in data:
+                if 'partner_id' not in data or not self._context.get('move_reverse_cancel'):
```